src/expr.rs

```diff
@@ -912,6 +912,15 @@ impl<'a> Rewrite for ControlFlow<'a> {
                                  block_str);
 
         if let Some(else_block) = self.else_block {
+            // Since this is an else block, we should not indent for the assignment preceding
+            // the original if, so set shape.indent.alignment to 0.
+            let shape = Shape {
+                width: shape.width,
+                indent: Indent {
+                    block_indent: shape.indent.block_indent,
+                    alignment: 0,
```

Comment:
Do you have an example in mind? Because as I understand it, the `} else if ... {` part will always move back to the block indent, which is retained.